Tests/FunctionalTests/MixedTargetTests.swift

```diff
@@ -0,0 +1,489 @@
+
+    func testMixedTargetWithInvalidCustomModuleMap() throws {
+        try fixture(name: "MixedTargets/BasicMixedTargets") { fixturePath in
+            // An invalid module map will cause the whole package to fail to
+            // build. To work around this, the module map is made invalid
+            // during the actual test.
+            let moduleMapPath = fixturePath.appending(
+                .init("Sources/MixedTargetWithInvalidCustomModuleMap/include/module.modulemap")
+            )
+
+            // In this case, an invalid module map is one that include a
+            // submodule of the form `$(ModuleName).Swift`. This is invalid
+            // because it collides with the submodule that SwiftPM will generate.
+            try """
+            module MixedTargetWithInvalidCustomModuleMap {
+                header "Foo.h"
+            }
+
+            module MixedTargetWithInvalidCustomModuleMap.Swift {}
+            """.write(to: moduleMapPath.asURL, atomically: true, encoding: .utf8)
+
+            let commandExecutionError = try XCTUnwrap(
+                XCTAssertBuildFails(
+                    fixturePath,
+                    extraArgs: ["--target", "MixedTargetWithInvalidCustomModuleMap"]
+                )
+            )
+
+            XCTAssert(
+                commandExecutionError.stderr.contains(
+                    "error: The target's module map may not contain a Swift " +
+                    "submodule for the module MixedTargetWithInvalidCustomModuleMap."
+                )
+            )
+        }
+    }
+
+    func testMixedTargetWithCustomModuleMapAndResources() throws {
+        try fixture(name: "MixedTargets/BasicMixedTargets") { fixturePath in
+            XCTAssertSwiftTest(
+                fixturePath,
+                extraArgs: [
+                    "--filter", "MixedTargetWithCustomModuleMapAndResourcesTests"
+// FIXME(ncooke3): Blocked by fix for #5728. Even though #5728 regression was
+// addressed in #6055, #5728 is guarded on Swift Tools Version `.vNext`– which
+// is also how the mixed language support is guarded. This comment should be
+// resolved once the mixed language test feature is staged for release and the
+// mixed language Fixture test targets have a swift-tools-version matching the
+// expected tools version that the mixed language test targets will release in.
+//                ],
+//                // Surface warning where custom umbrella header does not
+//                // include `resource_bundle_accessor.h` in `build` directory.
+//                Xswiftc: [
+//                    "-warnings-as-errors"
+                ]
+            )
+        } 
+    }
+
+    func testMixedTargetWithCXX_CXXInteropEnabled() throws {
+        try fixture(name: "MixedTargets/BasicMixedTargets") { fixturePath in
+            XCTAssertBuilds(
+                fixturePath,
+                extraArgs: ["--target", "MixedTargetWithCXX_CXXInteropEnabled"]
+            )
+        }
+    }
```

Comment:
@neonichu, CI [failed](https://ci.swift.org/job/swift-package-manager-with-xcode-self-hosted-PR-osx/2328/consoleFull) because CI is using an older toolchain that does not support the C++ interop flag I'm using in the target being built by this functional test: `-cxx-interoperability-mode=default. 

This isn't blocking, but is there way to configure CI's toolchain accordingly? Alternatively, I think I could guard the test with `#if swift(>=5.9)`.

Reply:
IMO guarding the test is the right thing to do.

Reply:
Makes sense, thanks @MaxDesiatov.